components/channel_header/channel_header.tsx

```diff
@@ -302,11 +318,8 @@ class ChannelHeader extends React.PureComponent {
 
         const channelNamesMap = channel.props && channel.props.channel_mentions;
 
-        let channelTitle = channel.display_name;
-        let archivedIcon = null;
-        if (channelIsArchived) {
-            archivedIcon = (<ArchiveIcon className='icon icon__archive icon channel-header-archived-icon svg-text-color'/>);
-        }
+        let channelTitle: ReactNode = channel.display_name;
```

Comment:
Do we really need to cast the type here? typically you don't have to use casting on initialized variables when they are assigned on init

Reply:
The reason I needed to cast is that:
- First I am assigning it to string (i.e. `channel.display_name`)
- Then it might be reassigned to `JSX.Element` (i.e. `FormattedMessage`, please see the code snippet below)
https://github.com/mattermost/mattermost-webapp/blob/e7806fd0b6a08ac3b9bf337051c883bfbd9819f0/components/channel_header/channel_header.tsx#L340-L348

This means that I need to cast the type as `JSX.Element | string` in the declaration of the variable, so it is easier for me to cast as `ReactNode` in the beginning.